extensions/sql-migration/src/dialog/skuRecommendationResults/GenerateProvisioningScriptDialog.ts

```diff
@@ -0,0 +1,223 @@
+
+	private dialog: azdata.window.Dialog | undefined;
+	private _isOpen: boolean = false;
+
+	private _disposables: vscode.Disposable[] = [];
+	private _armTemplateTextBox!: azdata.TextComponent;
+	private _armTemplateText!: string;
+
+	constructor(public model: MigrationStateModel, public _targetType: utils.MigrationTargetType) { }
+
+	private async initializeDialog(dialog: azdata.window.Dialog): Promise<void> {
+		return new Promise<void>((resolve, reject) => {
+			dialog.registerContent(async (view) => {
+				try {
+					const flex = this.CreateSaveArmTemplateContainer(view);
+
+					this._disposables.push(view.onClosed(e => {
+						this._disposables.forEach(
+							d => { try { d.dispose(); } catch { } });
+					}));
+
+					await view.initializeModel(flex);
+					resolve();
+				} catch (ex) {
+					reject(ex);
+				}
+			});
+		});
+	}
+
+	private CreateSaveArmTemplateContainer(_view: azdata.ModelView): azdata.FlexContainer {
+
+		const armTemplateDescription = _view.modelBuilder.text().withProps({
+			value: constants.TARGET_PROVISIONING_DESCRIPTION,
+			CSSStyles: {
+				...styles.BODY_CSS,
+				'margin-bottom': '8px',
+				'height': '36'
+			}
+		}).component();
+
+		this._armTemplateTextBox = _view.modelBuilder.text().withProps({
+			value: this._armTemplateText,
+			width: '100%',
+			height: '100%',
+			CSSStyles: {
+				'font': '12px "Monaco", "Menlo", "Consolas", "Droid Sans Mono", "Inconsolata", "Courier New", monospace',
+				'margin': '0',
+				'padding': '8px',
+				'white-space': 'pre',
+				'background-color': '#eeeeee',
+				'overflow-x': 'hidden',
+				'word-break': 'break-all'
+			}
+		}).component();
+
+		const textContainer = _view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'column',
+			height: 700,
+		}).withProps({
+			CSSStyles: {
+				'overflow': 'auto',
+				'user-select': 'text',
+				'margin-bottom': '8px',
+			}
+		}).withItems([
+			this._armTemplateTextBox
+		]).component();
+
+		const saveTemplateButton = _view.modelBuilder.button()
+			.withProps({
+				buttonType: azdata.ButtonType.Normal,
+				label: constants.TARGET_PROVISIONING_TITLE,
+				width: 117,
+				height: 36,
+				iconHeight: 16,
+				iconWidth: 16,
+				iconPath: IconPathHelper.openFolder,
+				CSSStyles: {
+					...styles.TOOLBAR_CSS
+				}
+			}).component();
+
+		saveTemplateButton.onDidClick(async () => {
+			const folder = await utils.promptUserForFolder();
+			if (folder) {
+				if (this._armTemplateText) {
+					let destinationFilePath: string | undefined;
+					switch (this._targetType) {
+						case utils.MigrationTargetType.SQLMI:
+							destinationFilePath = path.join(folder, 'ARMTemplate-AzureSqlManagedInstance' + new Date().toISOString().split('T')[0] + '.json');
+							break;
+
+						case utils.MigrationTargetType.SQLVM:
+							destinationFilePath = path.join(folder, 'ARMTemplate-AzureSqlVirtualMachine' + new Date().toISOString().split('T')[0] + '.json');
```

Comment:
Lets define this 'ARMTemplate-AzureSqlVirtualMachine' as constant

Reply:
used constant for target type.